### PR TITLE
Fix closing of DateBox when pressing "esc" key while editing time (T831957)

### PR DESCRIPTION
**Pull request by DokaRus**

### CODE REVIEW

File: js/ui/date_box/ui.time_view.js [hunk 1]
```diff
@@ -170,130 +176,149 @@ var TimeView = Editor.inherit({
...
+            editor.option("value", value);
+            editor._resumeValueChangeAction();
+        }
     },
 
     _getNumberBoxConfig: function() {
-        return extend({
+        return {
             showSpinButtons: true,
-            disabled: this.option("disabled"),
```
Comment (San4es) [inline js/ui/date_box/ui.time_view.js:266]: Why do we need this?

File: js/ui/date_box/ui.time_view.js [hunk 2]
```diff
@@ -303,11 +328,6 @@ var TimeView = Editor.inherit({
         }
     },
 
-    _toggleDisabledState: function(value) {
```
Comment (San4es) [inline js/ui/date_box/ui.time_view.js:306]: Why do we need to remove this method?

**Review by San4es — DISMISSED** [review]
LGTM

**Review by ksercs — DISMISSED** [review]

File: js/ui/date_box/ui.time_view.js (hunk 1 above)
Comment (DokaRus) [reply js/ui/date_box/ui.time_view.js:266]: When we create a nested widget via `createComponent` method, we say that root component should sync state of the `rtlEnabled` and `disabled` option with the nested one. So there is no need to set the initial value for this option

File: js/ui/date_box/ui.time_view.js (hunk 2 above)
Comment (DokaRus) [reply js/ui/date_box/ui.time_view.js:306]: This method is not necessary. To synchronize the `disabled` option, it is sufficient to create a nested component using the `createComponent` method

File: js/ui/date_box/ui.time_view.js (hunk 1 above)
Comment (San4es) [reply js/ui/date_box/ui.time_view.js:266]: Ok. Thank you!

File: js/ui/date_box/ui.time_view.js (hunk 2 above)
Comment (San4es) [reply js/ui/date_box/ui.time_view.js:306]: Ok.

**Review by alexander-kotov-dx — DISMISSED** [review]

**Review by San4es — APPROVED** [review]

**Review by alexander-kotov-dx — APPROVED** [review]

**Pull request merged**